systemd/emulation.go

```diff
@@ -147,12 +147,23 @@ func (s *emulation) AddMountUnitFile(snapName, revision, what, where, fstype str
 		return "", err
 	}
 
+	if modified == MountUnchanged {
+		return mountUnitName, nil
```

Comment:
in emulation I would expect that changing or preexisting units are unlikely but it still a good idea to support them, tests don't seem to reach the MountUnchanged and MountUpdated cases here

Reply:
I have added some tests.